### ISSUE TITLE
RNFetchBlob.config().fetch('GET', uri) always returns valid result

### ISSUE THREAD

**Issue by bdavid68** [wkh237/react-native-fetch-blob#492]
RNFetchBlob.config({}).fetch('GET', uri) function always returns valid result even though I am using invalid image url.(e.g. https://lol.com/lol.png)

React Version : "react": "16.0.0-alpha.6"
React Native Version : "react-native": "0.44.0"
Library Version : "react-native-fetch-blob": "0.10.2"

      return RNFetchBlob.config({
          path: getPath(uri),
      }).fetch('GET', uri);

      fetchImageUri('https://lol.com/lol.png')
         .then(result => {
           console.log('result = ', result);
         })
         .catch((error) => { console.log(error); });

This always returns valid local image url. So I can get the result of fetching image.
It never goes to *catch*.
How can I get the result of fetching image?

Thank you,

**Comment by lll000111** [comment]
What is `fetchImageUri`?

**Comment by bdavid68** [comment]
https://lol.com/lol.png

**Comment by lll000111** [comment]
Yes?

**Comment by bdavid68** [comment]
I don't see what you mean...

**Comment by lll000111** [comment]
Your code sample is incomplete. You return the promise created by the `fetch` to somewhere outside the screen...

**Comment by bdavid68** [comment]
Ah sorry.

        RNFetchBlob.config({
               path: getPath(uri),
            })
            .fetch('GET', 'https://lol.com/lol.png')
            .then(result => {
               console.log('result = ', result);
            })
            .catch((error) => { console.log(error); });
       }

**Comment by bdavid68** [comment]
Is there any solution?

**Comment by lll000111** [comment]
No. I wrote a test case for [the test package for RNFB](https://github.com/wkh237/react-native-fetch-blob-dev) for someone to continue from there.

```javascript
// REQUIRED TEST SETUP: Some tests, e.g. test-fetch.js, require this image
describe('#492 RNFetchBlob.fetch always resolves with valid result', (report, done) => {
  // 404 Not Found
  RNFetchBlob.fetch('GET', TEST_SERVER_URL + '/INVALID')
  .then(result => {
    report(
      <Assert key="Invalid URL should have caused error" expect={false} actual={true}/>,
      <Info key="Response data for fetch error test">
        <Text>{JSON.stringify(result)}</Text>
      </Info>
    )
    done()
  })
  .catch((error) => {
    report(
      <Assert key="Invalid URL causes error" expect={true} actual={true}/>,
      <Info key="Error">
        <Text>{String(error)}</Text>
      </Info>
    )
    done()
  });
})
```

Result:

![image](https://user-images.githubusercontent.com/25476357/29416484-85701306-8366-11e7-8290-43b5e7b72c91.png)

**Comment by bdavid68** [comment]
Hmm... This can't be the actual solution.

**Comment by lll000111** [comment]
Of course not. I said

> I wrote a test case for the test package for RNFB

I don't know how you expect bug fixing to work, the way I know it to work best is to have a test case that's part of ones existing test framework. If you look the RNFB test repo already contains numerous test cases specifically made for issues such es yours, starting with #[issue number]. I'm sorry if you don't see the value and expected an immediate resolution.

**Comment by wkh237** [comment]
@lll000111 , thanks for the great help 👍   let me run the test and figure out what's going on 😄

**Comment by wkh237** [comment]
@bdavid68 @lll000111  I think this is by designed. Quote from OkHttp's document 

> Note that transport-layer success (receiving a HTTP response code, headers and body) does not necessarily indicate application-layer success: response may still indicate an unhappy HTTP response code like 404 or 500.

So basically, you can check the `status` in  `res.info()` then deal with it, instead of try to catch the error.

**Comment by bdavid68** [comment]
Thanks @wkh237 !
I will try it.

**Comment by JulienMalige** [comment]
It seems already present ("react-native-fetch-blob": "0.10.8")
